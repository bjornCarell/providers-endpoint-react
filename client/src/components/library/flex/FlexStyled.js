import styled from 'styled-components';

export const FlexContainer = styled.div`
  display: flex;
  align-content: ${props => {
    if (props.alignContent) return props.content;
    if (props.contentStart) return 'flex-start';
    if (props.contentEnd) return 'flex-end';
    if (props.contentCenter) return 'center';
    if (props.contentBetween) return 'space-between';
    if (props.contentAround) return 'space-around';
    return 'stretch';
  }};
  align-items: ${props => {
    if (props.alignItems) return props.alignItems;
    if (props.alignStretch) return 'stretch';
    if (props.alignEnd) return 'flex-end';
    if (props.alignCenter) return 'center';
    if (props.alignBaseLine) return 'baseline';
    return 'flex-start';
  }};
  flex-direction: ${props => (props.column ? 'column' : 'row')};
  flex-wrap: ${props => {
    if (props.wrapReverse) return 'wrap-reverse';
    if (props.noWrap) return 'nowrap';
    return 'wrap';
  }};
  justify-content: ${props => {
    if (props.justifyContent) return props.justifyContent;
    if (props.justifyCenter) return 'center';
    if (props.justifyAround) return 'space-around';
    if (props.justifyBetween) return 'space-between';
    if (props.justifyEnd) return 'flex-end';
    return 'flex-start';
  }};
`;

export const FlexItem = styled.div`
  align-self: ${props => {
    if (props.aligSelf) return props.aligSelf;
    if (props.alignSelfEnd) return 'flex-end';
    if (props.alignSelfCenter) return 'center';
    return 'flex-start';
  }};
  flex: ${props => {
    if (props.flexNone) return 0;
    if (props.flexTwo) return 2;
    if (props.flexThree) return 3;
    return 1;
  }};
  padding: ${props => (props.noPadding ? 0 : '1.5rem')};
`;

export const FlexColumn = styled.div`
  width: ${props => {
    if (props.two) return '50%';
    if (props.three) return '33.33%';
    if (props.four) return '25%';
    return '100%';
  }};
`;
