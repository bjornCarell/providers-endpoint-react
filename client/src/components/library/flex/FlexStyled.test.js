import React from 'react';
import ReactDom from 'react-dom';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { FlexContainer } from './FlexStyled';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDom.render(<FlexContainer />, div);
});

it('renders default conditional styles correctly', () => {
  const tree = renderer
    .create(<FlexContainer>Providers</FlexContainer>)
    .toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('align-content', 'stretch');
  expect(tree).toHaveStyleRule('align-items', 'flex-start');
  expect(tree).toHaveStyleRule('flex-direction', 'row');
  expect(tree).toHaveStyleRule('justify-content', 'flex-start');
  expect(tree).toHaveStyleRule('flex-wrap', 'wrap');
});

it('renders conditional styles correctly', () => {
  const tree = renderer
    .create(
      <FlexContainer contentStart alignStretch column noWrap justifyCenter>
        Providers
      </FlexContainer>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('align-content', 'flex-start');
  expect(tree).toHaveStyleRule('align-items', 'stretch');
  expect(tree).toHaveStyleRule('flex-direction', 'column');
  expect(tree).toHaveStyleRule('justify-content', 'center');
  expect(tree).toHaveStyleRule('flex-wrap', 'nowrap');
});

it('renders conditional styles correctly', () => {
  const tree = renderer
    .create(
      <FlexContainer contentEnd alignEnd wrapReverse justifyAround>
        Providers
      </FlexContainer>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('align-content', 'flex-end');
  expect(tree).toHaveStyleRule('align-items', 'flex-end');
  expect(tree).toHaveStyleRule('flex-direction', 'row');
  expect(tree).toHaveStyleRule('justify-content', 'space-around');
  expect(tree).toHaveStyleRule('flex-wrap', 'wrap-reverse');
});

it('renders conditional styles correctly', () => {
  const tree = renderer
    .create(
      <FlexContainer contentCenter alignCenter justifyBetween>
        Providers
      </FlexContainer>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('align-content', 'center');
  expect(tree).toHaveStyleRule('align-items', 'center');
  expect(tree).toHaveStyleRule('justify-content', 'space-between');
});

it('renders conditional styles correctly', () => {
  const tree = renderer
    .create(
      <FlexContainer contentBetween alignBaseLine justifyEnd>
        Providers
      </FlexContainer>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule('align-content', 'space-between');
  expect(tree).toHaveStyleRule('align-items', 'baseline');
  expect(tree).toHaveStyleRule('justify-content', 'flex-end');
});
