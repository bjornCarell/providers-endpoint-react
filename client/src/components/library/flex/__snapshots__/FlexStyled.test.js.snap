// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders conditional styles correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: start;
  align-content: flex-start;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div
  className="c0"
>
  Providers
</div>
`;

exports[`renders conditional styles correctly 2`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: flex-end;
  -ms-flex-line-pack: end;
  align-content: flex-end;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap-reverse;
  -ms-flex-wrap: wrap-reverse;
  flex-wrap: wrap-reverse;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
}

<div
  className="c0"
>
  Providers
</div>
`;

exports[`renders conditional styles correctly 3`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

<div
  className="c0"
>
  Providers
</div>
`;

exports[`renders conditional styles correctly 4`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: space-between;
  -ms-flex-line-pack: space-between;
  align-content: space-between;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

<div
  className="c0"
>
  Providers
</div>
`;

exports[`renders default conditional styles correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

<div
  className="c0"
>
  Providers
</div>
`;
